serverlessworkflow/modules/ROOT/pages/persistence/persistence-with-postgresql.adoc

```diff
@@ -1 +1,166 @@
+
+.JDBC persistence add-on
+[source,xml]
+----
+<dependency>
+  <groupId>org.kie.kogito</groupId>
+  <artifactId>kogito-addons-quarkus-persistence-jdbc</artifactId>
+</dependency>
+----
+
+.Quarkus JDBC PostgreSQL
+[source,xml]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-jdbc-postgresql</artifactId>
+</dependency>
+----
+
+.Quarkus Agroal Data Source
+[source,xml]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-agroal</artifactId>
+</dependency>
+----
+
+[[postgresql-persistence-application-properties]]
+== Configure application.properties
+
+The following properties should be added to your project's application.properties file:
+
+.application.properties
+[source, properties]
+----
+kogito.persistence.type=jdbc
+----
+
+The following Quarkus properties are also recommended to be added:
+
+.application.properties
+[source, properties]
+----
+quarkus.datasource.db-kind=postgresql
+quarkus.datasource.username=postgres
+quarkus.datasource.password=pass
+quarkus.datasource.jdbc.url=jdbc:postgresql://localhost:5432/postgres
+----
+
+[[postgresql-persistence-database-schema]]
+== Create PostgreSQL database schema
+
+By default, the persistence addon creates the required database schema structure. That is usually only recommended for easy of development and can be turned off by setting `kogito.persistence.auto.ddl=false`.
+DDL scripts are available in case there is need to manually create the required database schema. These scripts are included in the https://repository.jboss.org/org/kie/kogito/kogito-ddl/{version}/kogito-ddl-{version}-db-scripts.zip[kogito-ddl-{version}-db-scripts.zip] artifact.
+
+
+[[postgresql-persistence-optimistic-locking]]
+== Enable optimistic locking
+
+In order to safely handle concurrent requests to shared workflow instances, Kogito handles these requests using persistence enabled optimistic locking for concurrency control with the version on record (using the version field in the database). This feature is optional and can be activated only with persistence by adding `kogito.persistence.optimistic.lock=true` to the application.properties file in your Kogito project.
+
+[[postgresql-persistence-versioning]]
+== Versioning strategy
+
+Versioning strategy can be used to allow different applications to run different versions of the same process at the same time, sharing the same database. This can be useful for instance, when migrating a process from version 1, to version 2. By allowing that workflow instances that still inflight to finish executing, while a new version is deployed using a new application setup. By default, the engine will consider that the `version` attribute specified in the Serverless Workflow file is the current version of the asset. This approach requires that you manually change this version, whenever you would like the engine to consider it as a new version.
```

Comment:
```suggestion
Versioning strategy can be used to allow different applications to run different versions of the same process at the same time, sharing the same database. This can be useful for instance, when migrating a process from version 1, to version 2. By allowing workflow instances that still inflight to finish executing, while a new version is deployed using a new application setup. By default, the engine will consider that the `version` attribute specified in the Serverless Workflow file is the current version of the asset. This approach requires that you manually change this version, whenever you would like the engine to consider it as a new version.
```